local-interchain/interchain/types/chains_test.go

```diff
@@ -0,0 +1,38 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestChainsGeneration(t *testing.T) {
+	t.Run("single chain runs", func(t *testing.T) {
+		require.NoError(t, ChainEthereum().SaveJSON("chains/gen-ethereum.json"))
+		require.NoError(t, ChainStargaze().SaveJSON("chains/gen-stargaze.json"))
+		require.NoError(t, ChainCosmosHub("localcosmos-1").SaveJSON("chains/gen-cosmoshub.json"))
+		require.NoError(t, ChainOsmosis().SaveJSON("chains/gen-osmosis.json"))
+		require.NoError(t, ChainJuno("localjuno-1").SaveYAML("chains/gen-juno.yml"))
```

Comment:
why is this the only one that uses yaml?

Reply:
Just to ensure it works as expected. If this failed the rest of them would fail due to how the logic is setup